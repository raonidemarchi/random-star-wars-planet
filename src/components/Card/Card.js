import React from 'react';
import styles from './Card.module.scss';

function Card({ children }) {
  return (
    <div className={styles.card}>
      {children}
    </div>
  );
}

function CardTitle({ children }) {
  return (
    <div className={styles.cardTitle}>
      {children}
    </div>
  );
}

function CardBody({ children }) {
  return (
    <div className={styles.cardBody}>
      {children}
    </div>
  );
}

function CardFooter({ children }) {
  return (
    <div className={styles.cardFooter}>
      {children}
    </div>
  );
}

export default Card;
export {
  CardTitle,
  CardBody,
  CardFooter
}