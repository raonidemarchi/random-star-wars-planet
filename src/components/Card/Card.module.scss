.card {
  max-width: 320px;
  width: 100%;
  background-color: #424242;
  border-radius: 2px;
  box-shadow: 0 16px 30px rgba(0, 0, 0, .2);
  text-align: center;

  .cardTitle {
    font-size: 34px;
    letter-spacing: 0.025rem;
    border-bottom: 1px solid #616161;
    padding: 24px;

    .placeholder {
      margin: 0 auto;
      max-width: 90%;
      min-height: 50px;
      border-radius: 2px;
      background-color: #616161;
    }

    .animatedBackground {
      animation-duration: 1.25s;
      animation-fill-mode: forwards;
      animation-iteration-count: infinite;
      animation-name: placeHolderShimmer;
      animation-timing-function: linear;
      background: darkgray;
      background: linear-gradient(to right, #616161 10%, #757575 18%, #616161 33%);
      background-size: 800px 104px;
      height: 50px;
      border-radius: 2px;
      position: relative;
    }
  }

  .cardBody {
    padding: 24px;

    .placeholder1 {
      margin: 0 auto;
      max-width: 70%;
      min-height: 14px;
      border-radius: 2px;
      background-color: #616161;
      margin-bottom: 10px;
    }

    .placeholder2 {
      @extend .placeholder1;
      max-width: 50%;
    }

    .placeholder3 {
      @extend .placeholder1;
      max-width: 90%;
      margin-bottom: 0;
    }

    .animatedBackground {
      animation-duration: 1.25s;
      animation-fill-mode: forwards;
      animation-iteration-count: infinite;
      animation-name: placeHolderShimmer;
      animation-timing-function: linear;
      background: darkgray;
      background: linear-gradient(to right, #616161 10%, #757575 18%, #616161 33%);
      background-size: 800px 104px;
      height: 14px;
      border-radius: 2px;
      position: relative;
    }
  }

  .cardFooter {
    font-size: 12px;
    letter-spacing: 0.040rem;
    padding: 8px 24px 24px;
  }
}

@keyframes placeHolderShimmer {
  0% {
      background-position: -468px 0
  }
  100%{
      background-position: 468px 0
  }
}