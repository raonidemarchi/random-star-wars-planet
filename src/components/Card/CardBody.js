import React from 'react';
import styles from './CardBody.module.scss';
import PlaceholderLoader from '../PlaceholderLoader/PlaceholderLoader';

function CardBody({ children, loading }) {
  return (
    <div className={styles.cardBody}>
      {
        loading ?
          <>
            <PlaceholderLoader height="16px" maxWidth="90%" marginBottom="10px" />
            <PlaceholderLoader height="16px" maxWidth="50%" marginBottom="10px" />
            <PlaceholderLoader height="16px" maxWidth="70%" />
          </>
        :
          <>
            {children}
          </>
      }
    </div>
  );
}

export default CardBody;
