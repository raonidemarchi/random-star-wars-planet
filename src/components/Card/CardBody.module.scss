.cardBody {
  padding: 24px;

  .placeholder1 {
    margin: 0 auto;
    max-width: 70%;
    min-height: 14px;
    border-radius: 2px;
    background-color: #616161;
    margin-bottom: 10px;
  }

  .placeholder2 {
    @extend .placeholder1;
    max-width: 50%;
  }

  .placeholder3 {
    @extend .placeholder1;
    max-width: 90%;
    margin-bottom: 0;
  }

  .animatedBackground {
    animation-duration: 1.25s;
    animation-fill-mode: forwards;
    animation-iteration-count: infinite;
    animation-name: placeHolderShimmer;
    animation-timing-function: linear;
    background: darkgray;
    background: linear-gradient(to right, #616161 10%, #757575 18%, #616161 33%);
    background-size: 800px 104px;
    height: 14px;
    border-radius: 2px;
    position: relative;
  }
}

@keyframes placeHolderShimmer {
  0% {
    background-position: -468px 0
  }
  100%{
    background-position: 468px 0
  }
}