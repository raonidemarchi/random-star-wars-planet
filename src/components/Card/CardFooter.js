import React from 'react';
import styles from './CardFooter.module.scss';
import PlaceholderLoader from '../PlaceholderLoader/PlaceholderLoader';

function CardFooter({ children, loading }) {
  return (
    <div className={styles.cardFooter}>
      {
        loading ?
          <>
            <PlaceholderLoader height="14px" maxWidth="50%" />
          </>
        :
          <>
            {children}
          </>
      }
    </div>
  );
}

export default CardFooter;
