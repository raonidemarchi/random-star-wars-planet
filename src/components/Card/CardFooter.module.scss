.cardFooter {
  font-size: 12px;
  letter-spacing: 0.040rem;
  padding: 8px 24px 24px;
}