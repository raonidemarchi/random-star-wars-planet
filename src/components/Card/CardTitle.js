import React from 'react';
import styles from './CardTitle.module.scss';

function CardTitle({ children, loading }) {
  return (
    <div className={styles.cardTitle}>
      {
        loading ?
          <>
            <div className={styles.placeholder}>
              <div className={styles.animatedBackground}></div>
            </div>
          </>
        :
          <>
            {children}
          </>
      }
    </div>
  );
}

export default CardTitle;
