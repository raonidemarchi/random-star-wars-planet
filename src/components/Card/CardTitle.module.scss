.cardTitle {
  font-size: 34px;
  letter-spacing: 0.025rem;
  border-bottom: 1px solid #616161;
  padding: 24px;
}