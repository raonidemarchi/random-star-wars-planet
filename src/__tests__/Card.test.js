import React from 'react';
import Card, { CardTitle, CardBody, CardFooter } from '../components/Card/Card';
import TestRenderer from 'react-test-renderer';

describe('Card component', () => {
  test('Should render a card with title, body and footer', () => {
    const tree = TestRenderer.create(
      <Card>
        <CardTitle>Title</CardTitle>
        <CardBody>Body</CardBody>
        <CardFooter>Footer</CardFooter>
      </Card>
    ).toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree.children[0].children).toStrictEqual(['Title']);
    expect(tree.children[1].children).toStrictEqual(['Body']);
    expect(tree.children[2].children).toStrictEqual(['Footer']);
  });
});