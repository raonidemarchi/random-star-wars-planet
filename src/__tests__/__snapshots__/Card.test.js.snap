// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Should render a card with title, body and footer 1`] = `
<div
  className="card"
>
  <div
    className="cardTitle"
  >
    Title
  </div>
  <div
    className="cardBody"
  >
    Body
  </div>
  <div
    className="cardFooter"
  >
    Footer
  </div>
</div>
`;
